src/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -170,14 +174,14 @@ protected int doWork() {
 
             // Figure out the coverage while not counting overlapping reads twice, and excluding various things
             final HashSet<String> readNames = new HashSet<String>(info.getRecordAndPositions().size());
+            int pileupSize=0;
             for (final SamLocusIterator.RecordAndOffset recs : info.getRecordAndPositions()) {
-                if (recs.getBaseQuality() < MINIMUM_BASE_QUALITY) {
-                    ++basesExcludedByBaseq;
-                    continue;
-                }
-                if (!readNames.add(recs.getRecord().getReadName())) {
-                    ++basesExcludedByOverlap;
-                    continue;
+
+                if (recs.getBaseQuality() < MINIMUM_BASE_QUALITY)                   { ++basesExcludedByBaseq;   continue; }
+                if (!readNames.add(recs.getRecord().getReadName()))                 { ++basesExcludedByOverlap; continue; }
+                pileupSize++;
+                if (pileupSize <= max) {
```

Comment:
If pileupsize>max, the base will not be counted (in the next step), and therefore I do not want to add its quality to the BQ distribution.